### PR TITLE
Feat/cfe 208 proposal description

### PR DESCRIPTION
**Pull request by jennieramida**
## Describe your changes
fix CFE-210

### CODE REVIEW

File: src/lib/pages/proposal-details/components/ProposalOverview.tsx [hunk 1]
```diff
@@ -27,21 +80,33 @@ export const ProposalOverview = () => {
...
+          </Flex>
+          <Flex direction="column" gap={4}>
+            <Heading as="h6" variant="h6">
+              Metadata
+            </Heading>
+            <Text
+              variant="body1"
+              color={proposalData.metadata.length ? "text.main" : "text.dark"}
+              wordBreak="break-word"
+            >
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/components/ProposalOverview.tsx:108]: combine with components in the renderMetadata fn

File: src/lib/pages/proposal-details/components/ProposalOverview.tsx [hunk 2]
```diff
@@ -27,21 +80,33 @@ export const ProposalOverview = () => {
...
-              service. Details can be seen in the Commonwealth thread:
-              https://gov.osmosis.zone/discussion/11973-skip-api-contract-upload
+            {proposalData.description.length === 0 ? (
+              <Text variant="body1" color="text.dark">
+                Not Provided
+              </Text>
+            ) : (
+              <Flex
+                direction="column"
+                gap={4}
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/components/ProposalOverview.tsx:90]: remove gap

File: src/lib/pages/proposal-details/components/ProposalOverview.tsx [hunk 3]
```diff
@@ -13,7 +16,57 @@ const ProposalStatus = () => {
...
   );
 };
-export const ProposalOverview = () => {
+
+const isUrl = (metadata: string): boolean => {
+  const urlRegex =
+    // eslint-disable-next-line no-useless-escape
+    /[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
+  return urlRegex.test(metadata);
+};
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/components/ProposalOverview.tsx:25]: new line

File: src/lib/pages/proposal-details/components/ProposalOverview.tsx [hunk 4]
```diff
@@ -13,7 +16,57 @@ const ProposalStatus = () => {
...
+interface ProposalOverviewProps {
+  proposalData: ProposalData;
+}
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/components/ProposalOverview.tsx:28]: new line

File: src/lib/components/__stories__/Markdown.stories.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    return fibonacci(n - 1) + fibonacci(n - 2);
+}
+
+for (let i = 0; i <= 10; i++) {
+    console.log(\`Fibonacci of \${i} is \${fibonacci(i)}\`);
+}
+\`\`\``;
+
+const list = `
+## Lists
```
Comment (songwongtp) [inline src/lib/components/__stories__/Markdown.stories.ts:61]: for consistency
"const list = `## Lists"

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/proposal-details/index.tsx [hunk 6]
```diff
@@ -83,7 +83,7 @@ const ProposalDetailsBody = ({
         </TabList>
         <TabPanels>
           <TabPanel p={0}>
-            <ProposalOverview />
+            <ProposalOverview proposalData={proposalData} />
```
Comment (evilpeach) [inline src/lib/pages/proposal-details/index.tsx:86]: data.info

File: src/lib/pages/proposal-details/components/ProposalOverview.tsx [hunk 7]
```diff
@@ -13,7 +16,57 @@ const ProposalStatus = () => {
     </Flex>
   );
 };
-export const ProposalOverview = () => {
+
+const isUrl = (metadata: string): boolean => {
+  const urlRegex =
+    // eslint-disable-next-line no-useless-escape
+    /[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
+  return urlRegex.test(metadata);
```
Comment (evilpeach) [inline src/lib/pages/proposal-details/components/ProposalOverview.tsx:24]: I think we can use Zod to determine whether input is a URL.

File: src/lib/utils/validate.ts [hunk 8]
```diff
@@ -27,6 +28,15 @@ export const isHex = (input: string): boolean => {
 export const isTxHash = (input: string): boolean =>
   isHex(input) && input.length === 64;
 
+export const isUrl = (string: string): boolean => {
```
Comment (songwongtp) [inline src/lib/utils/validate.ts:31]: Move this to line 17 (currently is in the middle of hex related fns)

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**